freeglut/freeglut/src/fg_internal.h

```diff
@@ -526,11 +579,14 @@ typedef void (*SFG_Proc)();
  * and for no other reason.  Since it's hidden in the macro, the
  * ugliness is felt to be rather benign.
  */
-#define SET_WCB(window,cbname,func)                            \
+#define SET_WCB(window,cbname,func,udata)                      \
 do                                                             \
 {                                                              \
     if( FETCH_WCB( window, cbname ) != (SFG_Proc)(func) )      \
```

Comment:
Sorry, i wasn't clear: i was referring to line 585, `if( FETCH_WCB( window, cbname ) != (SFG_Proc)(func) )`. If you call the setter again, only changing userdata, it'll be ignored. I don't see the need for separate setters either, that'd be hell.

I'm not sure how much the existing user data places are used. Guess on the user side you could have it point to a switchyard where you then select the right member function (if you're using userdata to enable using freeglut with member functions), this new implementation would be easier. You will see the docs for whats already there think of very simple scenarios only, not making living with freeglut from C++ easy. (I find, e.g., http://stackoverflow.com/questions/14865339/use-non-member-method-as-a-callback-in-glut, not easy and not flexible enough when you have more than one class to work with for instance. As such, i am for this addition. But perhaps it is best to see what others on the list say before going further.

Personally i also never read whole APIs, but i do search if i need something, and i would hope these functions would come up.